packages/wallets/talisman/src/helpers.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  Chain,
+  ChainToHexChainId,
+  addEVMWalletNetwork,
+  prepareNetworkSwitch,
+} from "@swapkit/helpers";
+
+import {
+  ARBToolbox,
+  BSCToolbox,
+  BrowserProvider,
+  ETHToolbox,
+  type Eip1193Provider,
+  MATICToolbox,
+  OPToolbox,
+} from "@swapkit/toolbox-evm";
+
+import { decodeAddress, encodeAddress } from "@polkadot/util-crypto";
+
+import { getToolboxByChain } from "@swapkit/toolbox-substrate";
+import type { InjectedWindow } from "./types";
+
+// Function to convert an address to a different format
+export const convertAddress = (inputAddress: string, newPrefix: number): string => {
+  const decodedAddress = decodeAddress(inputAddress);
+  const convertedAddress = encodeAddress(decodedAddress, newPrefix);
+  return convertedAddress;
+};
+
+export const getWeb3WalletMethods = async ({
+  ethereumWindowProvider,
+  chain,
+  covalentApiKey,
+  ethplorerApiKey,
+}: {
+  ethereumWindowProvider: Eip1193Provider | undefined;
+  chain: Chain;
+  covalentApiKey?: string;
+  ethplorerApiKey?: string;
+}) => {
+  if (!ethereumWindowProvider) throw new Error("Requested web3 wallet is not installed");
+
+  if (
+    (chain !== Chain.Ethereum && !covalentApiKey) ||
+    (chain === Chain.Ethereum && !ethplorerApiKey)
+  ) {
+    throw new Error(`Missing API key for ${chain} chain`);
+  }
+
+  const provider = new BrowserProvider(ethereumWindowProvider, "any");
+
+  const toolboxParams = {
+    provider,
+    signer: await provider.getSigner(),
+    ethplorerApiKey: ethplorerApiKey as string,
+    covalentApiKey: covalentApiKey as string,
+  };
+
+  const toolbox =
+    chain === Chain.Ethereum
+      ? ETHToolbox(toolboxParams)
+      : chain === Chain.Arbitrum
+        ? ARBToolbox(toolboxParams)
+        : chain === Chain.Polygon
+          ? MATICToolbox(toolboxParams)
+          : chain === Chain.Optimism
+            ? OPToolbox(toolboxParams)
+            : BSCToolbox(toolboxParams);
```

Comment:
switch case there plz + extract to separate function as it scales into infinite stairs and additionally - on chain which isn't "EVM" -> it will always resolve to BSC.

Also we have helper functions like `getToolboxForChain` - please use it :)

Reply:
fixed